muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.muzzle;
+
+import static java.util.Collections.emptyMap;
+
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.muzzle.ClassRef;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Map;
+
+/**
+ * This class is used to access {@code getMuzzleReferences} method from the {@link
+ * InstrumentationModule} class. That method is not visible in the source code and instead is
+ * generated automatically during compilation by the {@code
+ * io.opentelemetry.instrumentation.javaagent-codegen} Gradle plugin.
+ */
+class MuzzleReferencesAccessor {
+  private static final MethodHandle getMuzzleReferences;
+
+  static {
+    MethodHandles.Lookup lookup = MethodHandles.publicLookup();
+    MethodHandle handle;
+    try {
+      // This method is generated automatically during compilation by
+      // the io.opentelemetry.instrumentation.javaagent-codegen Gradle plugin.
+      //noinspection JavaLangInvokeHandleSignature
+      handle =
+          lookup.findVirtual(
+              InstrumentationModule.class, "getMuzzleReferences", MethodType.methodType(Map.class));
```

Comment:
The `InstrumentationModule` base class does not contain this method though, each of its implementations does - and they're all different methods (with same descriptor though) now that they don't override a common base method. Will this work properly?

Reply:
I will let our tests answer that...

Reply:
Seems like you were right :) Checking the new version...